theano/gof/type.py

```diff
@@ -634,7 +634,7 @@ def c_code(self, node, name, inputs, outputs, sub):
         """ % dict(ctype=self.rtype.ctype, out=outputs[0], inp=inputs[0])
 
     def c_code_cache_version(self):
-        return (0,)
+        return (0, self.rtype.version)
```

Comment:
In most ops that have such inner version, we do something like:

if inner_version is None:
    return None
return (N, inner_version)

I don't think we want that. Do you agree? If so, add a comment to talk about this.